src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -874,33 +875,13 @@ case class MatrixMapRows(child: MatrixIR, newRow: IR) extends MatrixIR {
   val tAgg = TAggregable(tAggElt, aggSymTab)
 
   val typ: MatrixType = {
-    val newRVRowSet = newRVRow.typ.fieldNames.toSet
-    val newRowKey = child.typ.rowKey.filter(newRVRowSet.contains)
-    val newPartitionKey = child.typ.rowPartitionKey.filter(newRVRowSet.contains)
+    val newRowKey = newKey.map{ case (pk, k) => pk ++ k }.getOrElse(child.typ.rowKey)
```

Comment:
how do you specify that the new key is actually None?

Reply:
This is MatrixMapRows, did you mean for TableMapRows? That's still handled with Table.unkey() and doesn't call into IR yet.